src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -336,69 +334,71 @@ def _check_remote_source_code(source_location):
-                return tarinfo
+        logger.debug(".dockerignore: no rule for '%s'.", tarinfo.name)
+        return tarinfo
 
-            for item in ignore_list:
-                if re.match(item.pattern, tarinfo.name):
-                    logger.debug(".dockerignore: rule '%s' matches '%s'.", item.rule, tarinfo.name)
-                    return None if item.ignore else tarinfo
+    with tarfile.open(tar_file_path, "w:gz") as tar:
+        # NOTE: Need to set arcname to empty string;
+        # otherwise the child item name will have a prefix (eg, ../) which can block unpacking.
+        tar.add(source_location, arcname="", filter=_filter_file)
 
-            logger.debug(
-                ".dockerignore: no rule for '%s'.", tarinfo.name)
-            return tarinfo
+    logger.debug("Starting to upload the archived source code from '%s'.", tar_file_path)
 
-        with tarfile.open(tar_file_path, "w:gz") as tar:
-            # NOTE: Need to set arcname to empty string;
-            # otherwise the child item name will have a prefix (eg, ../) which can block unpacking.
-            tar.add(source_location, arcname="", filter=_filter_file)
+    upload_url = None
+    error_message = "Could not get build source upload URL."
+    try:
+        source_upload_location = client.get_build_source_upload_url(resource_group_name, registry_name)
+        upload_url = source_upload_location.upload_url
+    except (AttributeError, CloudError) as e:
+        logger.debug("%s Exception: %s", error_message, e)
+        raise CLIError(error_message)
 
-        logger.debug(
-            "Starting to upload the archived source code from '%s'.", tar_file_path)
+    if not upload_url:
+        logger.debug("%s Empty build source upload URL.", error_message)
+        raise CLIError(error_message)
 
-        account_name, endpoint_suffix, container_name, blob_name, sas_token = _get_blob_info(
-            source_upload_location.upload_url)
+    account_name, endpoint_suffix, container_name, blob_name, sas_token = _get_blob_info(upload_url)
 
-        BlockBlobService(account_name=account_name, sas_token=sas_token,
+    try:
+        BlockBlobService(account_name=account_name,
+                         sas_token=sas_token,
                          endpoint_suffix=endpoint_suffix).create_blob_from_path(
-                             container_name=container_name, blob_name=blob_name, file_path=tar_file_path)
+                             container_name=container_name,
+                             blob_name=blob_name,
+                             file_path=tar_file_path)
 
         return source_upload_location.relative_path
     except Exception as err:
         try:
-            logger.debug(
-                "Starting to delete the archived source code from '%s'.", tar_file_path)
+            logger.debug("Starting to delete the archived source code from '%s'.", tar_file_path)
             os.remove(tar_file_path)
```

Comment:
The code doesn't look right to me. I remember I initially put the clean up in a finally block. Current code seems only doing the clean up when exception.

Reply:
Are you referring to this line https://github.com/AzureCR/azure-cli/blob/dev/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py#L283

I can track who made the change if needed. I am not touching the logic in this PR.